### DISCUSSION TITLE
Saleor Database Read Replica settings

### DISCUSSION THREAD

**Discussion by JannikZed** [saleor/saleor#10723] [Deployment]
Hi there,

We are evaluating the saleor read-replica features described in the docs here: https://docs.saleor.io/docs/3.x/developer/running-saleor/read-replicas

unfortunately, the docs are not making it clear to me, if read replicas work out-of-the-box for all core features of saleor, or if these docs are just informational for everybody trying to code new features into the core.

Our use-case: we would love to fine-tune the performance of saleor to always offer very good response times for all writing operations, by offloading all reading operations from the database master. We are expecting high traffic loads for a specific marketing event this year (~200 checkoutCreate / second) and want to prepare for that.

Any clarification on the read-replica feature is highly appreciated.

**Comment by fowczarek** [comment]
Hi @JannikZed, 

In Saleor we implement the usage of the read replicas(Mutation uses the primary database and resolvers use the replica).  You must prepare such replica and replication process in your environment by yourself.

**Reply by JannikZed** [reply]
Sounds great! I'm trying it out now, to see, if it brings us the expected results. I would like to make it run in our docker-based environment. the documentation is lacking any information on the needed configuration, so could you maybe provide an example on the needed environment settings to make read replication run? 
If I can make it run, I will update the docs via a PR later :)

**Reply by fowczarek** [reply]
From the Saleor perspective, you need to set config like in this [example.](https://docs.saleor.io/docs/3.x/developer/running-saleor/read-replicas#example-config).
The configuration of a replica and replication process depends on your environment.  You could do it differently on [AWS](https://aws.amazon.com/rds/features/read-replicas/), [GCP](https://cloud.google.com/sql/docs/postgres/replication/create-replica), [Azure](https://learn.microsoft.com/en-us/azure/postgresql/single-server/concepts-read-replicas), your local machine etc.. It's up to you how you handle it. I don't want to suggest any solution.
